src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/HACK_NodeHelper.cs

```diff
@@ -0,0 +1,136 @@
+        /// </summary>
+        private static void ExpandFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_ExpandFolder);
+            });
+        }
+
+        /// <summary>
+        /// Collapse a folder in a IVsUIHierarchy.
+        /// </summary>
+        private static void CollapseFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_CollapseFolder);
+            });
+        }
+
+        /// <summary>
+        /// Callbacks with a hierarchy and hierarchy window for use.
+        /// </summary>
+        private static void UseWindow(ConfiguredProject configuredProject, IServiceProvider serviceProvider, Action<IVsUIHierarchy, IVsUIHierarchyWindow> callback)
+        {
+            if (configuredProject.UnconfiguredProject.Services.HostObject is IVsUIHierarchy hierarchy)
+            {
+                callback(hierarchy, GetUIHierarchyWindow(serviceProvider, Guid.Parse(ManagedProjectSystemPackage.SolutionExplorerGuid)));
+            }
+        }
+
+        /// <summary>
+        /// Get reference to IVsUIHierarchyWindow interface from guid persistence slot.
+        /// </summary>
+        /// <param name="serviceProvider">The service provider.</param>
+        /// <param name="persistenceSlot">Unique identifier for a tool window created using IVsUIShell::CreateToolWindow.
+        /// The caller of this method can use predefined identifiers that map to tool windows if those tool windows
+        /// are known to the caller. </param>
+        /// <returns>A reference to an IVsUIHierarchyWindow interface, or <c>null</c> if the window isn't available, such as command line mode.</returns>
+        private static IVsUIHierarchyWindow GetUIHierarchyWindow(IServiceProvider serviceProvider, Guid persistenceSlot)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+
+            if (!(serviceProvider.GetService(typeof(SVsUIShell)) is IVsUIShell shell))
+            {
+                throw new InvalidOperationException();
+            }
+
+            object pvar = null;
+            IVsUIHierarchyWindow uiHierarchyWindow = null;
+
+            try
+            {
+                if (ErrorHandler.Succeeded(shell.FindToolWindow(0, ref persistenceSlot, out var frame)) && frame != null)
+                {
+                    ErrorHandler.ThrowOnFailure(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocView, out pvar));
+                }
+            }
+            finally
```

Comment:
Why is this in a failure block? Why do you ignore FindToolWindow, but throw on frame.GetProperty?

Reply:
You're trying to handle command-line mode? Under what circumstances would this code be called in command-line mode?

Reply:
I have no idea, this whole function was a copy and paste from CPS to get the hierarchy window.